x-pack/plugins/ml/public/application/components/model_snapshots/model_snapshots_table.tsx

```diff
@@ -0,0 +1,318 @@
+    },
+    {
+      field: 'timestamp',
+      name: i18n.translate('xpack.ml.modelSnapshotTable.time', {
+        defaultMessage: 'Date created',
+      }),
+      dataType: 'date',
+      render: renderDate,
+      sortable: true,
+    },
+    {
+      field: 'latest_record_time_stamp',
+      name: i18n.translate('xpack.ml.modelSnapshotTable.latestTimestamp', {
+        defaultMessage: 'Latest timestamp',
+      }),
+      dataType: 'date',
+      render: renderDate,
+      sortable: true,
+    },
+    {
+      field: 'retain',
+      name: i18n.translate('xpack.ml.modelSnapshotTable.retain', {
+        defaultMessage: 'Retain',
+      }),
+      width: '100px',
+      sortable: true,
+    },
+    {
+      field: '',
+      width: '100px',
+      name: i18n.translate('xpack.ml.modelSnapshotTable.actions', {
+        defaultMessage: 'Actions',
+      }),
+      actions: [
+        {
+          name: i18n.translate('xpack.ml.modelSnapshotTable.actions.revert.name', {
+            defaultMessage: 'Revert',
+          }),
+          description: i18n.translate('xpack.ml.modelSnapshotTable.actions.revert.description', {
+            defaultMessage: 'Revert to this snapshot',
+          }),
+          enabled: () => canCreateJob && canStartStopDatafeed,
+          type: 'icon',
+          icon: 'crosshairs',
+          onClick: checkJobIsClosed,
+        },
+        {
+          name: i18n.translate('xpack.ml.modelSnapshotTable.actions.edit.name', {
+            defaultMessage: 'Edit',
+          }),
+          description: i18n.translate('xpack.ml.modelSnapshotTable.actions.edit.description', {
+            defaultMessage: 'Edit this snapshot',
+          }),
+          enabled: () => canCreateJob,
+          type: 'icon',
+          icon: 'pencil',
+          onClick: setEditSnapshot,
+        },
+      ],
+    },
+  ];
+
+  if (snapshotsLoaded === false) {
+    return (
+      <>
+        <EuiFlexGroup justifyContent="spaceAround">
+          <EuiFlexItem grow={false}>
+            <EuiLoadingSpinner size="l" />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </>
+    );
+  }
+
+  return (
+    <>
+      <EuiInMemoryTable
+        className="eui-textOverflowWrap"
+        compressed={true}
+        items={snapshots}
+        columns={columns}
+        pagination={{
+          pageSizeOptions: [5, 10, 25],
+        }}
+        sorting={{
+          sort: {
+            field: 'timestamp',
+            direction: 'asc',
+          },
+        }}
+      />
+      {editSnapshot !== null && (
+        <EditModelSnapshotFlyout snapshot={editSnapshot} job={job} closeFlyout={closeEditFlyout} />
+      )}
+
+      {revertSnapshot !== null && (
+        <RevertModelSnapshotFlyout
+          snapshot={revertSnapshot}
+          snapshots={snapshots}
+          job={job}
+          closeFlyout={closeRevertFlyout}
+        />
+      )}
+
+      {closeJobModalVisible !== null && combinedJobState !== null && (
+        <EuiOverlayMask>
```

Comment:
Would be cleaner to extract this modal to dedicated components

Reply:
Changed in ac22c8ce86162fd9d4db4687850de4ae33873f4b